src/net.c

```diff
@@ -1535,12 +1535,58 @@ handle_request(coap_context_t *context, coap_queue_t *node) {
   assert(response == NULL);
 }
 
+static int
+coap_remove_separate_from_queue(coap_context_t *context, const coap_address_t *dst,
+			 const unsigned char *token, size_t token_length, coap_queue_t **sent) {
+  coap_queue_t *p, *q;
+  if (!context->sendqueue)
+    return 0;
+
+  if (coap_address_equals(dst, &context->sendqueue->remote) &&
+	    token_match(token, token_length,
+		              context->sendqueue->pdu->hdr->token,
+		              context->sendqueue->pdu->hdr->token_length) &&
+      context->sendqueue->retransmit_cnt > COAP_DEFAULT_MAX_RETRANSMIT) {
```

Comment:
Maybe we can encapsulate the condition ...retransmit_cnt > COAP_DEFAULT_MAX_RETRANSMIT in some way so that the semantics are easier to understand, e.g.:

```c
static inline int
is_separate(const coap_queue_t *node) {
   return node->retransmit_cnt > COAP_DEFAULT_MAX_RETRANSMIT;
}
```
and then:
```c
...coap_address_equals(....) && token_match(....) && is_separate(context->sendqueue)...
```